packages/runner/lib/exec.ts

```diff
@@ -18,7 +18,16 @@ export async function exec(
     codeParams?: object,
     abortController: AbortController = new AbortController()
 ): Promise<RunnerOutput> {
-    const rawNango = nangoProps.scriptType === 'action' ? new NangoAction(nangoProps) : new NangoSync(nangoProps);
+    const rawNango = (() => {
+        switch (nangoProps.scriptType) {
+            case 'sync':
+            case 'webhook':
+                return new NangoSync(nangoProps);
+            case 'action':
+            case 'post-connection-script':
+                return new NangoAction(nangoProps);
+        }
+    })();
```

Comment:
I don't think making `post-connection-script` sync was an explicit decision but the result of the code having a default value. If the ternary condition had been written differently I am sure we could have ended up with a `NangoAction`
That's the reason I am here using a switch. If we ever add a script type we would need to make a decision explicitely otherwise this code won't compile

Reply:
tbf Action is also wrong since we expose a lot of stuff that could not work before post-connection is executed, but we are close to rewrite the sdk 👍🏻

Reply:
one option is less wrong that the other 😜